### PR TITLE
#4073 updating kubeadm file.

### LINKED ISSUE #3915: Kubernetes v1.18.2 with crio version 1.18.2 failing to sync with kubelet on RH7
**Description**

I am relatively new to kubernetes. I can run my cluster when using the default socket (/var/run/dockershim.sock) but when I tried the crio socket to pull the images from my private repo I noticed the speed is not even close to compare with.

I am trying to configure all my nodes to use the crio.socket but I am failing to launch the master node with this socket.

I followed the documentation both from the kubernetes [Configuring each kubelet in your cluster using kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-control-plane-node) and also the git documentation [cri-o](https://github.com/cri-o/cri-o).

Unfortunately I am not able to get it working as it seems to be ignoring the private repo flag.

**Steps to reproduce the issue:**
1. Launch a master node (prime) with the following init (using a private repo):

```bash
kubeadm init \
        --upload-certs \
        --cri-socket=/var/run/crio/crio.sock \ 
        --node-name=my_node_name \
        --image-repository=my.private.repo \
        --pod-network-cidr=10.96.0.0/16 \
        --kubernetes-version=v1.18.2 \
        --control-plane-endpoint=ip:6443 \
        --apiserver-cert-extra-sans=ip \
        --apiserver-advertise-address=ip
```

2. Run as root or with sudo: ```journalctl -xeu crio -f```
3. Observe in debug or info mode the logs sample below

**Describe the results you received:**

Sample of logs from crio in debug mode:

```bash
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043499089+02:00" level=debug msg="Trying to access \"k8s.gcr.io/pause:3.2\"" file="docker/docker_image_src.go:68"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043547722+02:00" level=debug msg="Credentials not found" file="config/config.go:123"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043576124+02:00" level=debug msg="Using registries.d directory /etc/containers/registries.d for sigstore configuration" file="docker/lookaside.go:51"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043706369+02:00" level=debug msg=" Using \"default-docker\" configuration" file="docker/lookaside.go:169"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043736378+02:00" level=debug msg=" No signature storage configuration found for k8s.gcr.io/pause:3.2" file="docker/lookaside.go:174"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043769424+02:00" level=debug msg="Looking for TLS certificates and private keys in /etc/docker/certs.d/k8s.gcr.io" file="tlsclientconfig/tlsclientconfig.go:21"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.043858410+02:00" level=debug msg="GET https://k8s.gcr.io/v2/" file="docker/docker_client.go:516"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.046154250+02:00" level=debug msg="Ping https://k8s.gcr.io/v2/ err Get \"https://k8s.gcr.io/v2/\": dial tcp 10.254.3.15:443: connect: connection refused (&url.Error{Op:\"Get\", URL:\"https://k8s.gcr.io/v2/\", Err:(*net.OpError)(0xc00084d5e0)})" file="docker/docker_client.go:708"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.046239456+02:00" level=debug msg="GET https://k8s.gcr.io/v1/_ping" file="docker/docker_client.go:516"
Jun 30 20:03:45 hostname crio[6693]: time="2020-06-30 20:03:45.048653448+02:00" level=debug msg="Ping https://k8s.gcr.io/v1/_ping err Get \"https://k8s.gcr.io/v1/_ping\": dial tcp 10.254.3.15:443: connect: connection refused (&url.Error{Op:\"Get\", URL:\"https://k8s.gcr.io/v1/_ping\", Err:(*net.OpError)(0xc0006b0690)})" file="docker/docker_client.go:735"
```

**Describe the results you expected:**

Launching node with the use of crio socket

**Additional information you deem important (e.g. issue happens only occasionally):**

If I launch the node using the default socket e.g.:

```bash
# kubeadm init \
        --upload-certs \
        --cri-socket=/var/run/dockershim.sock \ 
        --node-name=my_node_name \
        --image-repository=my.private.repo \
        --pod-network-cidr=10.96.0.0/16 \
        --kubernetes-version=v1.18.2 \
        --control-plane-endpoint=ip:6443 \
        --apiserver-cert-extra-sans=ip \
        --apiserver-advertise-address=ip
W0630 20:24:33.223266   29033 configset.go:202] WARNING: kubeadm cannot validate component configs for API groups [kubelet.config.k8s.io kubeproxy.config.k8s.io]
[init] Using Kubernetes version: v1.18.2
[preflight] Running pre-flight checks
[preflight] Pulling images required for setting up a Kubernetes cluster
[preflight] This might take a minute or two, depending on the speed of your internet connection
[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
[kubelet-start] Writing kubelet environment file with flags to file "/var/lib/kubelet/kubeadm-flags.env"
[kubelet-start] Writing kubelet configuration to file "/var/lib/kubelet/config.yaml"
[kubelet-start] Starting the kubelet
[certs] Using certificateDir folder "/etc/kubernetes/pki"
[certs] Using existing ca certificate authority
[certs] Using existing apiserver certificate and key on disk
[certs] Using existing apiserver-kubelet-client certificate and key on disk
[certs] Using existing front-proxy-ca certificate authority
[certs] Using existing front-proxy-client certificate and key on disk
[certs] Using existing etcd/ca certificate authority
[certs] Using existing etcd/server certificate and key on disk
[certs] Using existing etcd/peer certificate and key on disk
[certs] Using existing etcd/healthcheck-client certificate and key on disk
[certs] Using existing apiserver-etcd-client certificate and key on disk
[certs] Using the existing "sa" key
[kubeconfig] Using kubeconfig folder "/etc/kubernetes"
[kubeconfig] Using existing kubeconfig file: "/etc/kubernetes/admin.conf"
[kubeconfig] Using existing kubeconfig file: "/etc/kubernetes/kubelet.conf"
[kubeconfig] Using existing kubeconfig file: "/etc/kubernetes/controller-manager.conf"
[kubeconfig] Using existing kubeconfig file: "/etc/kubernetes/scheduler.conf"
[control-plane] Using manifest folder "/etc/kubernetes/manifests"
[control-plane] Creating static Pod manifest for "kube-apiserver"
[control-plane] Creating static Pod manifest for "kube-controller-manager"
W0630 20:24:35.839949   29033 manifests.go:225] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
[control-plane] Creating static Pod manifest for "kube-scheduler"
W0630 20:24:35.841420   29033 manifests.go:225] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
[etcd] Creating static Pod manifest for local etcd in "/etc/kubernetes/manifests"
[wait-control-plane] Waiting for the kubelet to boot up the control plane as static Pods from directory "/etc/kubernetes/manifests". This can take up to 4m0s
[apiclient] All control plane components are healthy after 11.003647 seconds
[upload-config] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
[kubelet] Creating a ConfigMap "kubelet-config-1.18" in namespace kube-system with the configuration for the kubelets in the cluster
[upload-certs] Storing the certificates in Secret "kubeadm-certs" in the "kube-system" Namespace
[upload-certs] Using certificate key:
key
[mark-control-plane] Marking the node hostname as control-plane by adding the label "node-role.kubernetes.io/master=''"
[mark-control-plane] Marking the node hostname as control-plane by adding the taints [node-role.kubernetes.io/master:NoSchedule]
[bootstrap-token] Using token: token
[bootstrap-token] Configuring bootstrap tokens, cluster-info ConfigMap, RBAC Roles
[bootstrap-token] configured RBAC rules to allow Node Bootstrap tokens to get nodes
[bootstrap-token] configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
[bootstrap-token] configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
[bootstrap-token] configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
[bootstrap-token] Creating the "cluster-info" ConfigMap in the "kube-public" namespace
[kubelet-finalize] Updating "/etc/kubernetes/kubelet.conf" to point to a rotatable kubelet client certificate and key
[addons] Applied essential addon: CoreDNS
[addons] Applied essential addon: kube-proxy

Your Kubernetes control-plane has initialized successfully!

To start using your cluster, you need to run the following as a regular user:

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config

You should now deploy a pod network to the cluster.
Run "kubectl apply -f [podnetwork].yaml" with one of the options listed at:
  https://kubernetes.io/docs/concepts/cluster-administration/addons/

You can now join any number of the control-plane node running the following command on each as root:

  kubeadm join ip:6443 --token token \
    --discovery-token-ca-cert-hash sha256:hash \
    --control-plane --certificate-key key

Please note that the certificate-key gives access to cluster sensitive data, keep it secret!
As a safeguard, uploaded-certs will be deleted in two hours; If necessary, you can use
"kubeadm init phase upload-certs --upload-certs" to reload certs afterward.

Then you can join any number of worker nodes by running the following on each as root:

kubeadm join ip:6443 --token token \
    --discovery-token-ca-cert-hash sha256:hash
```

If I launch the node with crio socket:

```bash
# kubeadm init \
        --upload-certs \
        --cri-socket=/var/run/crio/crio.sock \ 
        --node-name=my_node_name \
        --image-repository=my.private.repo \
        --pod-network-cidr=10.96.0.0/16 \
        --kubernetes-version=v1.18.2 \
        --control-plane-endpoint=ip:6443 \
        --apiserver-cert-extra-sans=ip \
        --apiserver-advertise-address=ip
W0630 20:32:33.827957    2916 configset.go:202] WARNING: kubeadm cannot validate component configs for API groups [kubelet.config.k8s.io kubeproxy.config.k8s.io]
[init] Using Kubernetes version: v1.18.2
[preflight] Running pre-flight checks
[preflight] Pulling images required for setting up a Kubernetes cluster
[preflight] This might take a minute or two, depending on the speed of your internet connection
[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
[kubelet-start] Writing kubelet environment file with flags to file "/var/lib/kubelet/kubeadm-flags.env"
[kubelet-start] Writing kubelet configuration to file "/var/lib/kubelet/config.yaml"
[kubelet-start] Starting the kubelet
[certs] Using certificateDir folder "/etc/kubernetes/pki"
[certs] Generating "ca" certificate and key
[certs] Generating "apiserver" certificate and key
[certs] apiserver serving cert is signed for DNS names [hostname kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 10.96.134.57 10.96.134.57 10.96.134.57]
[certs] Generating "apiserver-kubelet-client" certificate and key
[certs] Generating "front-proxy-ca" certificate and key
[certs] Generating "front-proxy-client" certificate and key
[certs] Generating "etcd/ca" certificate and key
[certs] Generating "etcd/server" certificate and key
[certs] etcd/server serving cert is signed for DNS names [hostname localhost] and IPs [10.96.134.57 127.0.0.1 ::1]
[certs] Generating "etcd/peer" certificate and key
[certs] etcd/peer serving cert is signed for DNS names [hostname localhost] and IPs [10.96.134.57 127.0.0.1 ::1]
[certs] Generating "etcd/healthcheck-client" certificate and key
[certs] Generating "apiserver-etcd-client" certificate and key
[certs] Generating "sa" key and public key
[kubeconfig] Using kubeconfig folder "/etc/kubernetes"
[kubeconfig] Writing "admin.conf" kubeconfig file
[kubeconfig] Writing "kubelet.conf" kubeconfig file
[kubeconfig] Writing "controller-manager.conf" kubeconfig file
[kubeconfig] Writing "scheduler.conf" kubeconfig file
[control-plane] Using manifest folder "/etc/kubernetes/manifests"
[control-plane] Creating static Pod manifest for "kube-apiserver"
[control-plane] Creating static Pod manifest for "kube-controller-manager"
W0630 20:32:37.829806    2916 manifests.go:225] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
[control-plane] Creating static Pod manifest for "kube-scheduler"
W0630 20:32:37.830826    2916 manifests.go:225] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
[etcd] Creating static Pod manifest for local etcd in "/etc/kubernetes/manifests"
[wait-control-plane] Waiting for the kubelet to boot up the control plane as static Pods from directory "/etc/kubernetes/manifests". This can take up to 4m0s
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp [::1]:10248: connect: connection refused.
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp [::1]:10248: connect: connection refused.
Unfortunately, an error has occurred:
                timed out waiting for the condition

        This error is likely caused by:
                - The kubelet is not running
                - The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)

        If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
                - 'systemctl status kubelet'
                - 'journalctl -xeu kubelet'

        Additionally, a control plane component may have crashed or exited when started by the container runtime.
        To troubleshoot, list all containers using your preferred container runtimes CLI.

        Here is one example how you may list all Kubernetes containers running in cri-o/containerd using crictl:
                - 'crictl --runtime-endpoint /var/run/crio/crio.sock ps -a | grep kube | grep -v pause'
                Once you have found the failing container, you can inspect its logs with:
                - 'crictl --runtime-endpoint /var/run/crio/crio.sock logs CONTAINERID'

error execution phase wait-control-plane: couldn't initialize a Kubernetes cluster
To see the stack trace of this error execute with --v=5 or higher
``` 

I can see the localhost is listening on port 10248:
```bash
# curl -sSL http://localhost:10248/healthz
ok
```

Sample of crio socket (as described in documentation):
```bash
# curl -v --unix-socket /var/run/crio/crio.sock http://localhost/info | jq
* About to connect() to localhost port 80 (#0)
*   Trying /var/run/crio/crio.sock...
* Failed to set TCP_KEEPIDLE on fd 3
* Failed to set TCP_KEEPINTVL on fd 3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to localhost (/var/run/crio/crio.sock) port 80 (#0)
> GET /info HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Tue, 30 Jun 2020 18:36:35 GMT
< Content-Length: 240
<
{ [data not shown]
100   240  100   240    0     0   144k      0 --:--:-- --:--:-- --:--:--  234k
* Connection #0 to host localhost left intact
{
  "storage_driver": "overlay2",
  "storage_root": "/var/lib/containers/storage",
  "cgroup_driver": "systemd",
  "default_id_mappings": {
    "uids": [
      {
        "container_id": 0,
        "host_id": 0,
        "size": 4294967295
      }
    ],
    "gids": [
      {
        "container_id": 0,
        "host_id": 0,
        "size": 4294967295
      }
    ]
  }
}
```

**Output of `kubelet status`**
```bash
# systemctl status kubelet -l
● kubelet.service - kubelet: The Kubernetes Node Agent
   Loaded: loaded (/usr/lib/systemd/system/kubelet.service; enabled; vendor preset: disabled)
  Drop-In: /usr/lib/systemd/system/kubelet.service.d
           └─10-kubeadm.conf
   Active: active (running) since Tue 2020-06-30 20:39:49 CEST; 6s ago
     Docs: https://kubernetes.io/docs/
 Main PID: 8502 (kubelet)
    Tasks: 15
   Memory: 20.1M
   CGroup: /system.slice/kubelet.service
           └─8502 /usr/bin/kubelet --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --config=/var/lib/kubelet/config.yaml --container-runtime=remote --container-runtime-endpoint=/var/run/crio/crio.sock --hostname-override=hostname

Jun 30 20:39:55 hostname kubelet[8502]: I0630 20:39:55.369441    8502 kubelet_node_status.go:294] Setting node annotation to enable volume controller attach/detach
Jun 30 20:39:55 hostname kubelet[8502]: I0630 20:39:55.399015    8502 kubelet_node_status.go:70] Attempting to register node hostname
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.403707    8502 kubelet.go:2267] node "hostname" not found
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.503871    8502 kubelet.go:2267] node "hostname" not found
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.604115    8502 kubelet.go:2267] node "hostname" not found
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.704324    8502 kubelet.go:2267] node "hostname" not found
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.769448    8502 kubelet_node_status.go:92] Unable to register node "hostname" with API server: Post https://ip:6443/api/v1/nodes: dial tcp ip:6443: connect: connection refused
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.805779    8502 kubelet.go:2267] node "hostname" not found
Jun 30 20:39:55 hostname kubelet[8502]: E0630 20:39:55.906014    8502 kubelet.go:2267] node "hostname" not found
Jun 30 20:39:56 hostname kubelet[8502]: E0630 20:39:56.007272    8502 kubelet.go:2267] node "hostname" not found
```

From the little that I know the network errors is not relevant as I have not yet launched the network container, so the errors are expected at this point.

**Output of kubernetes containers with exit code**

```bash
# docker ps -a
CONTAINER ID        IMAGE                                     COMMAND                  CREATED             STATUS                      PORTS               NAMES
6b8382732b16        67da37a9a360                              "/coredns -conf /etc…"   20 minutes ago      Exited (0) 14 minutes ago                       k8s_coredns_coredns-7776bd96f-k8jtn_kube-system_de2a01fa-2bf1-4017-8b2a-bd8f592687cd_0
f95f9f087518        67da37a9a360                              "/coredns -conf /etc…"   20 minutes ago      Exited (0) 14 minutes ago                       k8s_coredns_coredns-7776bd96f-spxnm_kube-system_3dc2181a-9e28-49cf-8960-c491324f2f4a_0
9906ee71e4f6        0d40868643c6                              "/usr/local/bin/kube…"   20 minutes ago      Exited (2) 14 minutes ago                       k8s_kube-proxy_kube-proxy-xxgd2_kube-system_5253c388-a704-4d64-9831-5542e3558838_0
92d7ca86e75a        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_coredns-7776bd96f-spxnm_kube-system_3dc2181a-9e28-49cf-8960-c491324f2f4a_0
e7b2f4f660f9        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_coredns-7776bd96f-k8jtn_kube-system_de2a01fa-2bf1-4017-8b2a-bd8f592687cd_0
186dca4e580d        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_kube-proxy-xxgd2_kube-system_5253c388-a704-4d64-9831-5542e3558838_0
38ad48a696e0        a3099161e137                              "kube-scheduler --au…"   20 minutes ago      Exited (2) 14 minutes ago                       k8s_kube-scheduler_kube-scheduler-hostname_kube-system_73093ca80ee213950349d3c5398f795f_1
7e93d8af8a52        ace0a8c17ba9                              "kube-controller-man…"   20 minutes ago      Exited (2) 14 minutes ago                       k8s_kube-controller-manager_kube-controller-manager-hostname_kube-system_c78971ef2a4b13e277e268a33bdd029f_1
1b87828640aa        6ed75ad404bd                              "kube-apiserver --ad…"   20 minutes ago      Exited (0) 14 minutes ago                       k8s_kube-apiserver_kube-apiserver-hostname_kube-system_afaa823a7b3facc7d455c2ebdf1cf377_1
2ab274c32ba2        303ce5db0e90                              "etcd --advertise-cl…"   20 minutes ago      Exited (0) 14 minutes ago                       k8s_etcd_etcd-hostname_kube-system_46237b6d5c579bfc2e23d131ede859cb_1
1c290a566e64        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_etcd-hostname_kube-system_46237b6d5c579bfc2e23d131ede859cb_1
05eafe008864        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_kube-scheduler-hostname_kube-system_73093ca80ee213950349d3c5398f795f_1
a7aa2b6fad0a        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_kube-controller-manager-hostname_kube-system_c78971ef2a4b13e277e268a33bdd029f_1
a432eb2ed7ff        my.private.repo/pause:3.2   "/pause"                 20 minutes ago      Exited (0) 14 minutes ago                       k8s_POD_kube-apiserver-hostname_kube-system_afaa823a7b3facc7d455c2ebdf1cf377_1
```

If the containers are there why is failing? Since it is pulling the images why it is failing to launch the node?

**Output of `crio --version`:**

```bash
# crio --version
crio version 1.18.2
Version:       1.18.2
GitCommit:     7f261aeebffed079b4475dde8b9d602b01973d33
GitTreeState:  clean
BuildDate:     2020-06-18T21:05:27Z
GoVersion:     go1.14
Compiler:      gc
Platform:      linux/amd64
Linkmode:      static
```

**Output of `kubelet --version`:**

```bash
# kubelet --version
Kubernetes v1.18.2
```

**Output of `LinuxOS version`:**

```bash
# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.8 (Maipo)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

The installation is applied on barebone node.

### PR DESCRIPTION
**Pull request by thanos1983**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
Updating documentation on how to use CRI-O on offline networks.

#### Which issue(s) this PR fixes:

Fixes #3915

<!-
Fixes #3915 
or
None
-->

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: tutorials/kubeadm.md [hunk 1]
```diff
@@ -21,3 +21,47 @@ Note: This file assumes you've set your cgroup_driver as systemd
 
 Given you've set CIDR, and you've properly set the kubelet file, all you need to do is start crio (as defined [here](setup.md)), and run:
 `kubeadm init --pod-network-cidr=$CIDR`
+
+# Running kubeadm in an off line network
+
+We will assume that the user has installed CRI-O and alls necessary packages. We will also assume that all necessary components are configured and everything is working as expected. The user should have a private repo where the docker images are pushed. Sample of images fot version 1.18.2:
```
Comment (TomSweeneyRedHat) [inline tutorials/kubeadm.md:27]:
```suggestion
We will assume that the user has installed CRI-O and all necessary packages. We will also assume that all necessary components are configured and everything is working as expected. The user should have a private repo where the container images are pushed. An example of container images for Kubernetes version 1.18.2:
```

File: tutorials/kubeadm.md [hunk 2]
```diff
@@ -21,3 +21,47 @@ Note: This file assumes you've set your cgroup_driver as systemd
...
+user.private.repo/kube-apiserver:v1.18.2
+user.private.repo/kube-controller-manager:v1.18.2
+user.private.repo/kube-scheduler:v1.18.2
+user.private.repo/kube-proxy:v1.18.2
+user.private.repo/pause:3.2
+user.private.repo/etcd:3.4.3-0
+user.private.repo/coredns:1.6.7
+```
+
+The user needs to configure the [registries.conf](https://www.mankier.com/5/containers-registries.conf) file.
```
Comment (TomSweeneyRedHat) [inline tutorials/kubeadm.md:40]: I'd prefer pointing to the source, even though it's less pretty.
```suggestion
The user needs to configure the [registries.conf](https://github.com/containers/image/blob/master/docs/containers-registries.conf.5.md) file.
```

**Comment by TomSweeneyRedHat** [comment]
@thanos1983 thanks for the additional doc!  Looks good overall, I've just a few suggested tweaks.

File: tutorials/kubeadm.md (hunk 2 above)
Comment (thanos1983) [reply tutorials/kubeadm.md:40]: @TomSweeneyRedHat I agree with your comments. I did not noticed those small minor mistakes as well. Thank you for pointing them out :).

Let me know if there is anything else that I can help with.

BR / Thanos

**Comment by TomSweeneyRedHat** [comment]
LGTM
@thanos1983 it looks like the commit is missing a signature, but I don't recall this as a problem when I did the first review.  Did you perhaps forget the `-s` in your `git commit --amend -s` command?

File: tutorials/kubeadm.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ location = "user.private.repo"
 
 Next the user should reload and restart the CRI-O service to load the configurations.
 
-Last step. The user also should add the cgroup driver to the configuration file or pass it as a parameter when starting kubeadm for the master node.  The following are samples of parameters for the configuration file:
+The user should not forget to add the cgroup driver. The following are samples of parameters for the configuration file as documented also at [Configure cgroup driver used by kubelet on control-plane node](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-control-plane-node):
```
Comment (TomSweeneyRedHat) [inline tutorials/kubeadm.md:60]: This is not at all like the first review.  Did you mean to move to this?  I preferred the prior review fwiw.

Comment (thanos1983) [reply tutorials/kubeadm.md:60]: I will update it I though that the comments from @haircommander was to remove the whole line. I will revert.

Comment (thanos1983) [reply tutorials/kubeadm.md]: Done

**Comment by haircommander** [comment]
LGTM

**Comment by haircommander** [comment]
are you able to squash your commits into one and sign it off?

**Comment by thanos1983** [comment]
> are you able to squash your commits into one and sign it off?

Just pushed the last comment and signed it.

Let me know if there is something that I am missing.

BR / Thanos

**Comment by haircommander** [comment]
thanks @thanos1983 , though I think it would be better if you squashed all of the commits together. it seems some of the intermediate commits aren't signed off

**Comment by thanos1983** [comment]
> thanks @thanos1983 , though I think it would be better if you squashed all of the commits together. it seems some of the intermediate commits aren't signed off

Hello @haircommander,

You are right. Thank you for the note. I just pushed the full update including all comments. Hopefully now looks better. It is my first time that I am not pushing on one of my projects so I am having a bit of challenge. :)

**Comment by haircommander** [comment]
sorry, the commits are still unsigned and unsquashed. would you like me to take this over @thanos1983

**Comment by thanos1983** [comment]
> unsquashed

What do you mean by unsquashed? :)

**Comment by haircommander** [comment]
@thanos1983 carrying conversation from [here](https://github.com/cri-o/cri-o/pull/4073#issuecomment-681377105)

I see 7 commits in this PR, of which some are not signed. There should only be 1 commit, and it should be signed

**Comment by thanos1983** [comment]
> @thanos1983 carrying conversation from [here](https://github.com/cri-o/cri-o/pull/4073#issuecomment-681377105)
> 
> I see 7 commits in this PR, of which some are not signed. There should only be 1 commit, and it should be signed

I see the problem. I guess I can not update the commits from the UI so I need to revert to the master and push 1 commit from there.

Thanks for the guidance.

**Pull request closed without merging**